megamek/src/megamek/client/ui/swing/ClientGUI.java

```diff
@@ -460,12 +461,29 @@ public void initialize() {
 
         layoutFrame();
         menuBar.addActionListener(this);
+        frame.setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
         frame.addWindowListener(new WindowAdapter() {
             @Override
             public void windowClosing(WindowEvent e) {
-                frame.setVisible(false);
-                saveSettings();
-                die();
+                ignoreHotKeys = true;
+                int savePrompt = JOptionPane.showConfirmDialog(null,
+                        "Do you want to save the game before quitting MekHQ?",
```

Comment:
Maybe "before quitting MegaMek"?

Reply:
Ah, now everyone knows I just copied it from MekHQ... ok changed it and changed the icon to a warning icon.